### PR TITLE
Trino: support config timezone

### PR DESCRIPTION
**Pull request by duyet**
Support config trino timezone by providing it in connection extra field

```json
{ "timezone": "Asia/Ho_Chi_Minh" }
```

https://github.com/trinodb/trino-python-client?tab=readme-ov-file#timezone

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/trino/hooks/trino.py [hunk 1]
```diff
@@ -145,6 +145,7 @@ def get_conn(self) -> Connection:
             verify=_boolify(extra.get("verify", True)),
             session_properties=extra.get("session_properties") or None,
             client_tags=extra.get("client_tags") or None,
+            timezone=extra.get("timezone") or None,
```
Comment (Taragolis) [inline airflow/providers/trino/hooks/trino.py:148]: Should be also added into the [Documentation](https://airflow.apache.org/docs/apache-airflow-providers-trino/stable/connections.html):
- https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-trino/connections.rst

**Comment by Taragolis** [comment]
And also tests should be adopted to the changes

**Comment by eladkal** [comment]
@duyet can you please add a unit test?

**Comment by eladkal** [comment]
Closed in favor of https://github.com/apache/airflow/pull/35963

**Pull request closed without merging**